common/src/Model/WorldNode.h

```diff
@@ -53,6 +54,9 @@ namespace TrenchBroom {
             using NodeTree = AABBTree<FloatType, 3, Node*>;
             std::unique_ptr<NodeTree> m_nodeTree;
             bool m_updateNodeTree;
+
+            IdType m_nextPersistentGroupId = 0;
+            IdType m_nextPersistentLayerId = 0;
```

Comment:
In 2020.2 if you create four groups and three layers, the groups get ID's 1, 2, 3, 4, and the layers get ID's 5, 6, 7.

We should keep the 2020.2 ID generation behaviour (it's easier to debug since the ID's aren't shared) and have just one variable, m_nextPersistentId.

Also, the ID's need to start at 1 (2020.2 refuses to load ID 0).

Reply:
done! 8370ba53638ef0f1be99597ff3d7c362f7f001ad